test/functional/testlib.bash

```diff
@@ -1382,6 +1382,56 @@ set_latest_version() { # swupd_function
 
 }
 
+# Creates a new version of the server side content in the specified
+# location to serve as a 3rd-party repository
+# Parameters:
+# - ENVIRONMENT_NAME: the name of the test environment
+# - VERSION: the version of the server side content
+# - FORMAT: the format to use for the version
+# - REPO_NAME: the name of the 3rd-party repo, defaults to "test-repo"
+create_third_party_repo() { #swupd_function
+
+	local env_name=$1
+	local version=$2
+	local format=${3:-staging}
+	local repo_name=${4:-test-repo}
+	local repo_state_dir
+	local path
+
+	# If no parameters are received show usage
+	if [ $# -eq 0 ]; then
+		cat <<-EOM
+			Usage:
+			    create_third_party_repo <environment_name> <new_version> [format] [repo_name]
+			EOM
+		return
+	fi
+	validate_item "$env_name"
+	validate_param "$version"
+
+	debug_msg "Creating 3rd-party repo $repo_name"
+	path=$(dirname "$(realpath "$env_name")")
+
+	# create the state dir for the 3rd-party repo
+	repo_state_dir="$STATEDIR"/3rd-party/"$repo_name"
+	debug_msg "Creating a state directory for repo $repo_name at $repo_state_dir"
+	sudo mkdir -p "$repo_state_dir"/{staged,download,delta,telemetry,bundles}
+	sudo chmod -R 0700 "$repo_state_dir"
+	export REPO_STATEDIR="$repo_state_dir"
+
+	# create the basic content for the 3rd-party repo
+	create_version "$env_name" "$version" 0 "$format" "$repo_name"
+	debug_msg "3rd-party repo $repo_name created successfully"
+
+	# add the new repo to the repo.ini file
+	{
+		printf '[%s]\n\n' "$repo_name"
```

Comment:
I think this should be a single \n

Reply:
we always have a blank line between sections and key/values -> https://github.com/clearlinux/swupd-client/blob/master/config